tools/compile_tf_graph.py

```diff
@@ -159,34 +210,86 @@ def post_compile(cls, rec_layer_name, network, output_file_name=None):
+      # dependency on init_base_op for each init_op
+      with tf.control_dependencies([init_base_op]):  
+        init_ops_list.append(var.init_op())
+      info["collections"]["encode_ops"].append(var.init_op().name)
+      # choice-dependent loop vars are not I/O stored, but need to be updated after feeding choices
+      if var.choice_dependent:
+        update_ops_list.append(var.final_op())
+        info["collections"]["update_ops"].append(var.final_op().name)
+      # update loop vars after the last decode op
+      elif var.choice_dependent is False:
+        post_update_ops_list.append(var.final_op())
+        info["collections"]["post_update_ops"].append(var.final_op().name)
+
+    # group encode_ops #
+    with tf.control_dependencies( [init_base_op] ):
+      init_op = tf.group(*init_ops_list, name="rec_step_by_step_init_op")
+    encode_ops_coll.append( init_op )
+ 
+    # group post_update_ops #
+    post_update_op = tf.group(*post_update_ops_list, name="rec_step_by_step_post_update_op")
+    post_update_ops_coll.append( post_update_op )
+
+    # stochastic vars # 
     print("Stochastic vars, and their order:")
+    tile_batch_multiple = None
     for name in rec_layer.stochastic_var_order:
       print(" %s" % name)
       info["stochastic_var_order"].append(name)
       info["stochastic_vars"][name] = {
         "calc_scores_op": rec_layer.state_vars["stochastic_var_scores_%s" % name].final_op().name,
         "scores_state_var": "stochastic_var_scores_%s" % name,
         "choice_state_var": "stochastic_var_choice_%s" % name}
+      # collect decode_input_vars
+      input_var = rec_layer.state_vars["stochastic_var_choice_%s" % name].var
+      decoder_input_vars_coll.append(input_var)
+      info["collections"]["decoder_input_vars"].append(input_var.name)
+      # collect decode_output_vars
+      output_var = rec_layer.state_vars["stochastic_var_scores_%s" % name].var
+      decoder_output_vars_coll.append(output_var)
+      info["collections"]["decoder_output_vars"].append(output_var.name)
+      # automatic tile encodings w.r.t. choices
+      if tile_batch_multiple is None:
+        tile_batch_multiple = rec_layer.state_vars["stochastic_var_choice_%s" % name].read_batch_dim()
+      # collect decode_ops
+      decode_op = rec_layer.state_vars["stochastic_var_scores_%s" % name].final_op()
+      decode_ops_coll.append(decode_op)
+      info["collections"]["decode_ops"].append(decode_op.name)
+
+    # automatic tile encodings w.r.t. choices
+    tile_op = cell.set_parent_tile_multiple_op(tile_batch_multiple)
+    update_ops_list.append(tile_op)
+    info["collections"]["update_ops"].append(tile_op.name)
+    
+    # group update ops
+    update_op = tf.group(*update_ops_list, name="rec_step_by_step_update_op")
+    update_ops_coll.append( update_op )
+
     import json
     info_str = json.dumps(info, sort_keys=True, indent=2)
-    print("JSON:")
-    print(info_str)
```

Comment:
I intentionally had this here, always, just to have some nicer output, to see the structure, what is being saved.

Reply:
it's very annoying everytime full screen output. output file can be checked to see the nice output